action/protocol/context.go

```diff
@@ -63,6 +68,36 @@ type (
 		// Nonce is the nonce of the action
 		Nonce uint64
 	}
+
+	// CheckFunc is function type to check by height.
+	CheckFunc func(height uint64) bool
+
+	// FeatureCtx provides features information.
+	FeatureCtx struct {
+		DepositGasLast            bool
+		SystemWideActionGasLimit  bool
+		NotFixTopicCopyBug        bool
+		SetRevertMessageToReceipt bool
+		FixGetHashFnHeight        bool
+		UsePendingNonceOption     bool
+		AsyncContractTrie         bool
+		StoreOutOfGasToReceipt    bool
+		RefundAllDeposit          bool
+		AddChainIDToConfig        bool
+		UseV2Storage              bool
+		CheckUnstaked             bool
+		SkipStakingIndexer        bool
+		ReturnFetchError          bool
+		CanNotTranferToSelf       bool
+		PostFairbankMigration     bool
+	}
+
+	// FeatureWithHeightCtx provides feature check functions.
+	FeatureWithHeightCtx struct {
+		GetUnproductiveDelegates CheckFunc
+		EnableSMStorage          CheckFunc
+		ReadStateFromDB          CheckFunc
```

Comment:
we could move into `FeatureCtx`?
so no need for an extra `FeatureWithHeightCtx`